### DISCUSSION TITLE
Assigning Pore Size Distribution and Porosity to Network

### DISCUSSION THREAD

**Discussion by Thraiye** [PMEAL/OpenPNM#2792] [Q&A]
Hello OpenPNM community,

I have been working on the code for predicting CO2-brine relative permeability using statistical data for reconstructing the network. However, there are a few things I could not figure out even after searching in the discussion forum:

1) Assigning a specific porosity value to the network

2) Generating pore and throat size distributions as the ones I assigned

I shall share a file with my code for reference, hopefully, someone can guide me on what I may be missing in the code to get accurate results. It does not give any error, but the results generated are not corresponding to what I assigned.

Thanks in advance for your guidance.

[Thraiye CO2-Brine Rel Perm.docx](https://github.com/PMEAL/OpenPNM/files/12156634/Thraiye.CO2-Brine.Rel.Perm.docx)

**Comment by jgostick** [comment]
Hi @Thraiye 
1. You cannot assign a porosity value.  You assign pore/throat sizes, and spacing, then porosity is fixed.  If you're pore/throat sizes are good, then can tweak the porosity by adjust their spacing, with wider spacing giving lower porosity.  
2. For adjust the PSD, here is a tutorial that might be helpful: https://openpnm.org/examples/applications/mercury_intrusion.html
